### DISCUSSION TITLE
Waybar still visible behind transparent windows

### DISCUSSION THREAD

**Discussion by pinarruiz** [hyprwm/Hyprland#8757] [Q&A] [answered]
Hi to everyone,

I have recently made the switch to hyprland and was configuring some nice trasnparency + blur. I have configured Alacritty to have trasnparency of its own so that the characters dont get "dimmer", the issue is that my waybar is still visible behind fullscreen windows of alacritty:

https://github.com/user-attachments/assets/a802fc64-44f2-418c-9dac-f94d5ca1b9ee

As you can see on the video it hides other windows properly, but it keeps waybar there.
I was going to start doing something with [IPC](https://wiki.hyprland.org/IPC/) but figured I'd ask here first as that seems a bit extreme for this.

Thanks in advance for any suggestion.

Edit: Alacritty is the one setting its own transparency, I am not using the opacity from Hyprland.

**Comment by vaxerski** [comment, marked as answer]
what layer is your bar on? Doesn't happen to me with qs on layer bottom

edit: ah, you probably have blur optimizations off (why?)

not possible to have it gone atm unless you put waybar on the top layer. You can make a feature request.

**Reply by pinarruiz** [reply]
>  you probably have blur optimizations off (why?)

I am starting on hyprland and wayland, coming from Qtile (X11), so the config is pretty "new". As far as i know they come enabled by default and i have not disabled them (Just learning about that now).

My hyprland version info is:

```
Hyprland 0.45.2 built from branch v0.45.2-b at commit 12f9a0d0b93f691d4d9923716557154d74777b0a  ([gha] Nix: update inputs).
Date: 2024-11-19
Tag: v0.45.2, commits: 12f9a0d0b93f691d4d9923716557154d74777b0a
built against aquamarine 0.4.4
```

And the related config is:
```conf
decoration {
  blur {
    enabled=true
    passes=1
    size=3
    vibrancy=0.169600
  }

  shadow {
    color=rgba(1a1a1aee)
    enabled=true
    range=4
    render_power=3
  }
  active_opacity=1.000000
  inactive_opacity=1.000000
  rounding=5
}

layerrule=blur, waybar
layerrule=blurpopups, waybar
layerrule=ignorealpha, waybar

windowrulev2=suppressevent maximize, class:.*
windowrulev2=nofocus,class:^$,title:^$,xwayland:1,floating:1,fullscreen:0,pinned:0
windowrulev2=noanim, class:^(flameshot)$
windowrulev2=float, class:^(flameshot)$
windowrulev2=move 0 0, class:^(flameshot)$
windowrulev2=pin, class:^(flameshot)$
windowrulev2=opacity 0.900000 0.900000, class:^(org.pulseaudio.pavucontrol|io.missioncenter.MissionCenter|.blueman-manager-wrapped|com.nextcloud.desktopclient.nextcloud|org.gnome.Nautilus|Rquickshare|.virt-manager-wrapped|io.gitlab.adhami3310.Impression)$
windowrulev2=noborder, floating:0, onworkspace:w[tv1]
windowrulev2=noborder, floating:0, onworkspace:f[1]
windowrulev2=norounding, floating:0, onworkspace:w[tv1]
windowrulev2=norounding, floating:0, onworkspace:f[1]
windowrulev2=noshadow, floating:0, onworkspace:w[tv1]
windowrulev2=noshadow, floating:0, onworkspace:f[1]

workspace=w[tv1], gapsout:0, gapsin:0
workspace=f[1], gapsout:0, gapsin:0
```

**Reply by pinarruiz** [reply]
In the end setting the layer of waybar to `top` fixed the issue, thanks.

**Reply by atoxey** [reply]
As it is not working in the bottom layer right now you can add this workaround to your hyprland.conf to make it work.

`bind = $mainMod, <Fullscreen Key>, exec, killall -SIGUSR1 waybar`

Sadly, if you have more monitors, waybar would not be visible while in fullscreen in one of them.
